packages/commercetools-sdk-middleware-auth/src/build-requests.js

```diff
@@ -12,34 +12,34 @@ type BuiltRequestParams = {
   body: string;
 }
 
-const defaultAuthHost = 'https://auth.sphere.io'
-
 // POST https://{host}/oauth/token?grant_type=client_credentials&scope={scope}
 // Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==
 export function buildRequestForClientCredentialsFlow (
-  {
-    host = defaultAuthHost,
-    projectKey,
-    credentials,
-    scopes,
-  }: AuthMiddlewareOptions = {},
+  options: AuthMiddlewareOptions,
 ): BuiltRequestParams {
-  if (!projectKey)
+  // TODO: use a better validator
```

Comment:
Yes and no :) For example, the `credentials` is an object, and we need to validate that both keys in it are provided.

My question was mostly if we do this manually (array or not array) or have some kind of schema validator.